tests/suites/test_suite_ccm.data

```diff
@@ -1517,3 +1517,143 @@ mbedtls_ccm_encrypt_and_tag:MBEDTLS_CIPHER_ID_CAMELLIA:"D75C2778078CA93D971F96FD
+
+CCM* encrypt, overflow update NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_overflow_update:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_ENCRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* encrypt, incomplete update NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_incomplete_update:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_ENCRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* encrypt, full update and overflow NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_full_update_and_overflow:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_ENCRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, overflow ad NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_overflow_ad:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, incomplete ad NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_incomplete_ad:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, full ad and overflow NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_full_ad_and_overflow:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, overflow update NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_overflow_update:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, incomplete update NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_incomplete_update:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
+CCM* decrypt, full update and overflow NIST VADT AES-192 #14 (P=24, N=13, A=13, T=16)
+depends_on:MBEDTLS_AES_C
+mbedtls_ccm_full_update_and_overflow:MBEDTLS_CIPHER_ID_AES:MBEDTLS_CCM_STAR_DECRYPT:"b5f43f3ae38a6165f0f990abe9ee50cd9ad7e847a0a51731":"3726c1aaf85ee8099a7ebd3268700e07d4b3f292c65bba34":"13501aebda19a9bf1b5ffaa42a":"ead4c45ff9db54f9902a6de181"
+
```

Comment:
Missing good cases: not calling `update_ad` at all (with A=0); not calling either `update_ad` or `update` (with A=0 and P=0). I.e. similar to `mbedtls_ccm_skip_update`, but skipping `update_ad`, or skipping both.

Reply:
Please be more specific here. Calling `update` and `update_ad` with 0 input is covered by `check_multipart` function, `n1` and `n1_add` are iterated from 0 to `n`.

Reply:
What's covered is calling `update_ad(0); update(0);`. But **not calling the functions** is not covered. We need to cover all of the following call sequences (where `a`, `p` are `> 0`):
```
{
    start();
    set_lengths(a, p);
    update_ad(a);
    update(p);
    finish();
}

{
    start();
    set_lengths(0, p);
    update_ad(0);
    update(p);
    finish();
}

{
    start();
    set_lengths(0, p);
    // no update_ad
    update(p);
    finish();
}

{
    start();
    set_lengths(a, 0);
    update_ad(a);
    update(0);
    finish();
}

{
    start();
    set_lengths(a, 0);
    update_ad(a);
    // no update
    finish();
}

{
    start();
    set_lengths(0, 0);
    update_ad(0);
    update(0);
    finish();
}

{
    start();
    set_lengths(0, 0);
    // no update_ad
    update(0);
    finish();
}

{
    start();
    set_lengths(0, 0);
    update_ad(0);
    // no update
    finish();
}

{
    start();
    set_lengths(0, 0);
    // no update_ad
    // no update
    finish();
}
```

This can be done with separate functions or by leveraging the existing data and adding additional calls to `check_multipart`which skip the `update_ad` call, skip the `update` call, or both.

Reply:
```
{
    start();
    set_lengths(a, p);
    update_ad(a);
    update(p);
    finish();
} - covered by mbedtls_ccm_encrypt_and_tag and mbedtls_ccm_auth_decrypt tests, such as:
CCM encrypt and tag NIST VNT AES-192 #1 (P=24, N=7, A=32, T=16)
CCM auth decrypt tag NIST DVPT AES-128 #27 (P=24, N=7, A=32, T=16)

{
    start();
    set_lengths(0, p);
    update_ad(0);
    update(p);
    finish();
} - covered by: mbedtls_ccm_encrypt_and_tag and mbedtls_ccm_auth_decrypt tests, such as:
CCM encrypt and tag NIST VADT AES-128 #1 (P=24, N=13, A=0, T=16)
CCM auth decrypt tag NIST DVPT AES-128 #11 (P=24, N=7, A=0, T=16)

{
    start();
    set_lengths(0, p);
    // no update_ad
    update(p);
    finish();
} - covered by mbedtls_ccm_skip_ad, such as:
CCM encrypt, skip ad NIST VADT AES-128 (P=24, N=13, A=0, T=16)
CCM decrypt, skip ad NIST DVPT AES-192 (P=24, N=7, A=0, T=4)

{
    start();
    set_lengths(a, 0);
    update_ad(a);
    update(0);
    finish();
} - covered by: mbedtls_ccm_encrypt_and_tag and mbedtls_ccm_auth_decrypt tests, such as:
CCM encrypt and tag NIST VPT AES-128 #1 (P=0, N=13, A=32, T=16)
CCM auth decrypt tag NIST DVPT AES-128 #17 (P=0, N=7, A=32, T=4)

{
    start();
    set_lengths(a, 0);
    update_ad(a);
    // no update
    finish();
} - covered by mbedtls_ccm_skip_update tests, such as:
CCM encrypt, skip update NIST VPT AES-128 #1 (P=0, N=13, A=32, T=16)
CCM decrypt, skip update NIST DVPT AES-256 #23 (P=0, N=13, A=32, T=16)

{
    start();
    set_lengths(0, 0);
    update_ad(0);
    update(0);
    finish();
} - covered by mbedtls_ccm_auth_decrypt tests for P=0, A=0
such as: CCM auth decrypt tag NIST DVPT AES-128 #1 (P=0, N=7, A=0, T=4)
Missing: mbedtls_ccm_encrypt_and_tag tests for P=0, A=0

{
    start();
    set_lengths(0, 0);
    // no update_ad
    update(0);
    finish();
} - missing mbedtls_ccm_skip_ad tests for P=0, A=0

{
    start();
    set_lengths(0, 0);
    update_ad(0);
    // no update
    finish();
} - missing mbedtls_ccm_skip_update tests for P=0, A=0

{
    start();
    set_lengths(0, 0);
    // no update_ad
    // no update
    finish();
} - missing mbedtls_ccm_instant_finish P=0, A=0
```
so:
- [ ] add mbedtls_ccm_encrypt_and_tag tests for P=0, A=0
- [ ] add mbedtls_ccm_skip_ad tests for P=0, A=0
- [ ] add mbedtls_ccm_skip_update tests for P=0, A=0
- [ ] add mbedtls_ccm_instant_finish P=0, A=0

Reply:
Can't find [here](https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/cavp-testing-block-cipher-modes#CCM) a proper test vector (P=0, A=0) for mbedtls_ccm_encrypt_and_tag test

Reply:
If you can't find a test vector matching some conditions from NIST or an RFC, you can build one with a well-established implementation such as python/cryptodome or openssl.